backend/pikturr/test/args-tests.js

```diff
@@ -0,0 +1,256 @@
+          if (fs.statSync(filePath).isFile())
+            fs.unlinkSync(filePath);
+          else
+            rmDir(filePath);
+        }
+    fs.rmdirSync(dirPath);
+};
+
+
+before(function(){
+    console.log('Setting up test/tmp directory.')
+    // let's create a temp folder where (almost) all test artifacts will be created
+    // reset its content in this method to start the test suite in a know state
+
+    // remove any old test artifacts
+    rmDir(testTempDir)
+    // make sure full path exist
+    mkdirp.sync(testTempDir)
+
+    // create an empty file
+    // a test will try to create a sub-dir on it and will fail (as expected)
+    fs.closeSync(fs.openSync(testNonDirPath, 'w'));
+})
+
+
+
+describe("Missing UML Argument", function() {
+      it("call index.js without a URL argument", function() {
+        missingUrl()
+  });
+});
+
+describe("Fail if output path cannot be created", function() {
+      it("call index.js with an impossible output path", function() {
+        failMkdirp()
+  });
+});
+
+describe("Generate UML File", function() {
+      it("call index.js requesting to generate a plantuml file", function() {
+        genUmlFile()
+  });
+});
+
+describe("Generate PNG File", function() {
+      it("call index.js requesting to generate a png file", function() {
+        // this is a slower test. it generates a whole png file
+        // on my MacBook Pro it took about 2.5 seconds
+        // bumping this up to 10 seconds just in case
+        // adjust if needed in the future
+        this.timeout(10000);
+        genPngFile()
+  });
+});
+
+describe("Generate UML File in Deep Directory", function() {
+      it("call index.js requesting a plantuml file, in a deep dir", function() {
+        genUmlFileDeepDir()
+  });
+});
+
+
+describe("Generate UML File with default name", function() {
+      it("call index.js requesting a plantuml file, name it from the input file/URL", function() {
+        genUmlFileDefaultName()
+  });
+});
+
+
+var missingUrl = function () {
+    // call index.js without a URL to read swagger specs from
+    // the process should exit with a return code of 1 for this case
+    var prc = spawnSync('node', ['index.js', '--uml'])
+//    console.log("prc.status = " + prc.status)
+//    console.log("prc.signal = " + prc.signal)
+//    console.log("prc.error = " + prc.error)
+//    console.log("prc.stdout = " + prc.stdout)
+//    console.log("prc.stderr = " + prc.stderr)
+
+    exptErr = "Last argument must be the path or url of the swagger file to read."
+
+    expect(prc.status, "exit code is 1").to.equal(1);
+    expect(prc.stderr.toString('utf-8')).to.have.string(exptErr);
+}
+
+var failMkdirp = function () {
+    // call index.js requesting a path that cannot be created (e.g. one of the dir is a file)
+    // the process should exit with a return code of 2 for this case
+    var prc = spawnSync('node', ['index.js', '-o', testImpossibleDirPath, '--uml', 'http://petstore.swagger.io/v2/swagger.yaml'])
```

Comment:
SonarCloud is complaining about the http request as it is not safe. Try https instead and see if that works without creating problems.

Reply:
Sonarcloud [link](https://sonarcloud.io/project/security_hotspots?id=purplesmurf1998_Autotracks&pullRequest=56&resolved=false&types=SECURITY_HOTSPOT) here

Reply:
I deleted the args-tests.js file in pikturr because we can create a diagram without it. I added the commit and sonarCloud passes it.

Reply:
Nice work.